src/realm/spec.cpp

```diff
@@ -309,6 +309,14 @@ void Spec::insert_column(size_t column_ndx, ColKey col_key, ColumnType type, Str
 
     m_types.insert(column_ndx, int(type)); // Throws
     m_attr.insert(column_ndx, attr); // Throws
+    if (type != col_type_BackLink) {
+        if (auto ref = m_top.get_as_ref(3)) {
+            Array coll_types(m_top.get_alloc());
+            coll_types.set_parent(&m_top, 3);
```

Comment:
This is general comment, I saw in the code we do this basically everywhere, we basically use these magic numbers to set up parents or references inside the file. It would be great to maybe refactor this at some point, in order to map all those indexes into some constants. 
In this case, it is not clear that we are actually setting the parent for the collection owned and setting also the position of this collection in the parent array.

Reply:
Yes, we should refactor to use constants like in Table and Group.